### PR TITLE
.Net: Add a sample which shows how to use a function filter to modify the request payload

### PR DESCRIPTION
**Pull request by markwallace-microsoft**
### Motivation and Context

Shows how to create a filter which will modify the `KernelArguments` before a function is called.

This sample does the following:

1. Registers a FunctionInvocationFilter
2. Handles "createRepair" function calls by
    1. Updating the payload to set the Assigned To and the Date 

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/IntegrationTests/Plugins/RepairServiceTests.cs [hunk 1]
```diff
@@ -115,15 +222,47 @@ public class Repair
...
+                    repair.AssignedTo = "Karin Blair";
+                    context.Arguments["payload"] = JsonSerializer.Serialize(repair);
+                }
+            }
+
+            await next(context);
+        }
+    }
+
+    public sealed class CustomHandler(HttpMessageHandler innerHandler) : DelegatingHandler(innerHandler)
```
Comment (RogerBarreto) [inline dotnet/src/IntegrationTests/Plugins/RepairServiceTests.cs:257]: Make it private, as this class is being used only within this test class

```suggestion
    private sealed class CustomHandler(HttpMessageHandler innerHandler) : DelegatingHandler(innerHandler)
```

File: dotnet/src/IntegrationTests/Plugins/RepairServiceTests.cs [hunk 2]
```diff
@@ -106,7 +108,112 @@ public async Task HttpOperationExceptionIncludeRequestInfoAsync()
...
+        // List All Repairs
+        result = await plugin["listRepairs"].InvokeAsync(kernel);
+
+        Assert.NotNull(result);
+        repairs = JsonSerializer.Deserialize<Repair[]>(result.ToString());
+        Assert.True(repairs?.Length > 0);
+        Assert.Equal(count + 2, repairs?.Length);
+    }
+
+    public sealed class Repair
```
Comment (RogerBarreto) [inline dotnet/src/IntegrationTests/Plugins/RepairServiceTests.cs:216]: Make it private, as this class is being used only within this test class

```suggestion
    private sealed class Repair
```

**Review by RogerBarreto — APPROVED** [review]

**Comment by RogerBarreto** [comment]
@markwallace-microsoft this change actually is adding an Integration Test and not a Sample in the Concepts. Update the description or add the sample and `readme.md` with this example.

File: dotnet/src/IntegrationTests/Plugins/RepairServiceTests.cs [hunk 3]
```diff
@@ -1,7 +1,11 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
```
Comment (westey-m) [inline dotnet/src/IntegrationTests/Plugins/RepairServiceTests.cs:1]: This file is in integration tests. Should it be in samples?

File: dotnet/src/IntegrationTests/Plugins/RepairServiceTests.cs [hunk 4]
```diff
@@ -115,15 +279,83 @@ public class Repair
...
+        {
+            // Modify the HttpRequestMessage
+            request.Headers.Add("CustomHeader", "CustomValue");
+
+            // Call the next handler in the pipeline
+            return await base.SendAsync(request, cancellationToken);
+        }
+    }
+
+    public sealed class FunctionMessageHandler(HttpMessageHandler innerHandler) : DelegatingHandler(innerHandler), IFunctionInvocationFilter
```
Comment (westey-m) [inline dotnet/src/IntegrationTests/Plugins/RepairServiceTests.cs:326]: If we are adding a sample, shouldn't it be in samples?

**Pull request closed without merging**